flowexperimental/comp/wells/SingleCompWellState.hpp

```diff
@@ -0,0 +1,117 @@
+/*
+  Copyright 2024, SINTEF Digital
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef OPM_SINGLE_COMP_WELL_STATE_HPP
+#define OPM_SINGLE_COMP_WELL_STATE_HPP
+
+#include <opm/input/eclipse/Schedule/Well/WellEnums.hpp>
+
+#include <string>
+#include <vector>
+
+#include <opm/simulators/wells/PerforationData.hpp>
+
+namespace Opm {
+
+template <typename Scalar>
+class ConnectionData
```

Comment:
Do we intend to unify this with the existing [`PerfData`](https://github.com/OPM/opm-simulators/blob/45663690f5172d3a529942c40a5ba18976e38d5a/opm/simulators/wells/PerfData.hpp#L33) type?  If not, I foresee potential confusion as to which type goes where.

Reply:
We intend to unify with the existing `PerfData`.  The `PerfData` in its current form has too many things that we are not sure whether we want to handle even in the far future and does not have the variables needed for compositional simulation. 

We can refactor so that they unify based on base class, while the PR is trying to avoid introducing changes in the main flow at the current stage.

Reply:
I rename `ConnectionData` to `CompConnectionData` hopefully it is less confusing.